### PR TITLE
Added hdr parser.

### PR DESCRIPTION
**Pull request by figo2264**
Fixes #

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

I've added HdrParser from playcanvas viewer to the original engine.

Just wonder why this good feature is not included.

### CODE REVIEW

File: src/resources/texture.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ import { BasisParser } from './parser/texture/basis.js';
 import { ImgParser } from './parser/texture/img.js';
 import { KtxParser } from './parser/texture/ktx.js';
 import { LegacyDdsParser } from './parser/texture/legacy-dds.js';
+import { HdrParser} from "./parser/texture/hdr";
```
Comment (yaustar) [inline src/resources/texture.js:15]:
```suggestion
import { HdrParser } from "./parser/texture/hdr";
```

Comment (yaustar) [reply src/resources/texture.js:15]: Lint fix

**Comment by willeastcott** [comment]
@slimbuck will review this on Monday. 😄

**Review by slimbuck — CHANGES_REQUESTED** [review]
Hi @figo2264 and thanks for this PR!

@willeastcott, just checking you're happy for us to make this HDR parser part of the engine? It would mean we can remove the two versions we have (in the viewer and engine examples), but will increase engine size.

File: src/resources/parser/texture/hdr.js [hunk 2]
```diff
@@ -0,0 +1,234 @@
+import { http } from '../../../net/http.js';
+
+
+function Reader(data) {
```
Comment (willeastcott) [inline src/resources/parser/texture/hdr.js:4]: Can you make this an ES6 class please?

**Comment by willeastcott** [comment]
I'm OK adding this to the engine if you are @slimbuck. It's not huge and it is being used by the viewer. In future, I'm hoping tree shaking will make it less relevant how big the engine is anyway.

**Comment by Maksims** [comment]
Will it be possible to add support for HDR textures in Editor with help of this PR?

**Comment by willeastcott** [comment]
@Maksims That's actually non-trivial, because currently, HDR images are auto-converted to RGBM PNGs on import into the Editor. So we'd need to add support for HDR import options that allow them to be treated as a runtime format.

**Comment by slimbuck** [comment]
Closing this PR in favour of #2973, which makes a few additional changes.

**Pull request closed without merging**